client/authority.go

```diff
@@ -0,0 +1,71 @@
+package client
+
+import (
+	"context"
+
+	pb "github.com/go-phorce/trusty/api/v1/trustypb"
+	"google.golang.org/grpc"
+)
+
+type authorityClient struct {
+	remote   pb.AuthorityClient
+	callOpts []grpc.CallOption
+}
+
+// NewAuthority returns instance of Authority client
+func NewAuthority(conn *grpc.ClientConn, callOpts []grpc.CallOption) Authority {
+	return &authorityClient{
+		remote:   RetryAuthorityClient(conn),
+		callOpts: callOpts,
+	}
+}
+
+// NewAuthorityFromProxy returns instance of Authority client
+func NewAuthorityFromProxy(proxy pb.AuthorityClient) Authority {
+	return &authorityClient{
+		remote: proxy,
+	}
+}
+
+// ProfileInfo returns the certificate profile info
+func (c *authorityClient) ProfileInfo(ctx context.Context, in *pb.CertProfileInfoRequest) (*pb.CertProfileInfo, error) {
+	return c.remote.ProfileInfo(ctx, in, c.callOpts...)
+}
+
+// CreateCertificate returns the certificate
+func (c *authorityClient) CreateCertificate(ctx context.Context, in *pb.CreateCertificateRequest) (*pb.CertificateBundle, error) {
+	return c.remote.CreateCertificate(ctx, in, c.callOpts...)
+}
+
+// Issuers returns the issuing CAs
+func (c *authorityClient) Issuers(ctx context.Context) (*pb.IssuersInfoResponse, error) {
+	return c.remote.Issuers(ctx, emptyReq, c.callOpts...)
+}
+
+type retryAuthorityClient struct {
```

Comment:
`type retryAuthorityClient struct {` seem to be very much the same as `type authorityClient struct {`.

Most likely in the future, we just need one client who does retries. I do not think we need two structures for this. 

Is it possible to implement retry logic within `github.com/go-phorce/trusty/api/v1/trustypb` itself (I guess the answer is no because it is an auto-generated code)?